apply_factors_wrangler.py

```diff
@@ -132,9 +132,28 @@ def lambda_handler(event, context):
             on=distinct_values,
             how="inner",
         )
-
         logger.info("Successfully merged non-responders with factors")
 
+        # Region/strata combinations that exist in responder data
+        # but not non_responders get dropped off on join
+        # With factors. Identify these, merge on the regionless factor.
+        # Then concat onto original dataset.
+        # It looked a lot nicer before flake8....
+        dropped_rows = non_responder_dataframe_with_prev[
+            ~non_responder_dataframe_with_prev['responder_id'].isin(
+                non_responders_with_factors['responder_id'])].dropna()
+        if(len(dropped_rows) > 0):
+            row14factors = \
+                factors_dataframe[
+                    produce_columns("imputation_factor_", questions_list, distinct_values)
+                ][factors_dataframe['region'] == 14]
```

Comment:
There is a 'regionless_code' runtime var which should be used instead of hard coding '14'. Removing '14' from the variable name is also advised.